Modules/Sources/WordPressCoreProtocols/CachedAndFetchedResult.swift

```diff
@@ -0,0 +1,21 @@
+import Foundation
+
+public protocol CachedAndFetchedResult<T>: Sendable {
```

Comment:
(nit) This seems a bit confusing. What is `UncachedResult` for? Why does `result` have two async closures? Is it more like a promise?

Reply:
Yeah, it's technically two promises – one for fetching from the cache and one for fetching from the network.

It allows you to return a single object from an endpoint that makes it easy to read the same type from the cache quickly then from the network. 

Might need a better name, but it's a lot more ergonomic to work with from view code than `AsyncStream<T>`